src/libopal.c

```diff
@@ -2875,7 +2875,43 @@ gen_bin (char *obj_fn, char *dest_fn)
   /// Assert destination file name is not null
   assert(dest_fn);
 
-  // TODO
+  /// Call to linker
+  logger(DEBUG, "Calling LD to link object.");
+  char LD_cmd[1024] = { 0 };
+  sprintf(LD_cmd, "ld -m elf_x86_64 -o %s -lc -I/lib64/ld-linux-x86-64.so.2 %s", dest_fn, obj_fn);
+
+  /// Confirm obj_fn can be read
+  logger(DEBUG, "access(obj_fn, R_OK)");
+  errno = EXIT_SUCCESS;
+  if (access(obj_fn, R_OK) == 0){
+
+      /// and confirm dest_fn can be written to and executed
+      logger(DEBUG, "access(obj_fn, W_OK) && access(obj_fn, X_OK)");
+      errno = EXIT_SUCCESS;
+      if ((access(dest_fn, W_OK) == 0) && (access(dest_fn, W_OK) == 0)){
```

Comment:
You should check if dest_fn can be read and executed **after** ld is run.
We want `ld` to create a file with the name `dest_fn` and then check if it was created correctly by testing for the `rwx` bits.

Right now, it will fail at this point, since the dest_fn doesn't exist yet.